.github/workflows/tests-on-pr-push-and-merge.yml

```diff
@@ -70,12 +70,33 @@ jobs:
         env:
           TEST_USER_SA_KEY: ${{ secrets.TEST_USER_SA_KEY }}
           DEV_CI_SA_KEY: ${{ secrets.DEV_CI_SA_KEY }}
+      - name: Build Docker image
+        id: build_docker_image
+        run: |
+          # additionally pull the main branch for on-push-to-PRs, so we can diff and see what changed
+          if [ "$GHA_EVENT_NAME" = "pull_request" ]; then
+            git fetch --no-tags --depth=1 origin main
```

Comment:
You might want to use `grep '^docker/'`

Reply:
`git diff` starts the line with a letter indicating the type of change (e.g. `M` for modified). e.g.
```
> git diff --name-status origin/main
M       .github/workflows/tests-on-pr-push-and-merge.yml
M       docker/scripts/terra_init.sh
```
I'll change it to `grep '\sdocker'` instead, so it matches on the space before the file path.

Reply:
ah sorry...  you'd need to also use `--name-only` instead of `--name-status`...  Either way works for me...

Reply:
Ah makes sense. I didn't think to check for different output formats. Thanks for the pointer!